pypeit/core/findobj_skymask.py

```diff
@@ -147,7 +147,1159 @@ def create_skymask(sobjs, thismask, slit_left, slit_righ, box_rad_pix=None, trim
+                hand_extract_dict=hand_extract_dict,  
+                nperslit=nperorder, extract_maskwidth=extract_maskwidth,
+                snr_thresh=snr_thresh, trim_edg=trim_edg, 
+                boxcar_rad=box_radius/plate_scale_ord[iord],
+                show_peaks=show_peaks, show_fits=show_single_fits,
+                show_trace=show_single_trace, qa_title=qa_title, specobj_dict=specobj_dict,
+                objfindQA_filename=ech_objfindQA_filename)
+        sobjs.add_sobj(sobjs_slit)
+
+    # Return
+    return sobjs
+
+
+def ech_fof_sobjs(sobjs:specobjs.SpecObjs, 
+                  slit_left:np.ndarray, 
+                  slit_righ:np.ndarray, 
+                  plate_scale_ord:np.ndarray, 
+                  fof_link:float=1.5):
+    """
+    Links together objects previously found using a 
+    friends-of-friends algorithm on fractional order position.
+
+    Each source from each order is then assigned an obj_id value
+
+    Args:
+        sobjs (:class:`~pypeit.specobj.SpecObj`):
+            Previously found objects
+        slit_left (`numpy.ndarray`_):
+            Left boundary of orders to be extracted (given as floating point
+            pixels).  Shape is (nspec, norders), where norders is the total
+            number of traced echelle orders.
+        slit_righ (`numpy.ndarray`_):
+            Right boundary of orders to be extracted (given as floating point
+            pixels).  Shape is (nspec, norders), where norders is the total
+            number of traced echelle orders.
+        plate_scale_ord (`numpy.ndarray`_):
+            An array with shape (norders,) providing the plate 
+            scale of each order in arcsec/pix, 
+        fof_link (:obj:`float`, optional):
+            Friends-of-friends linking length in arcseconds used to link
+            together traces across orders. The routine links together at
+            the same fractional slit position and links them together
+            with a friends-of-friends algorithm using this linking
+            length.
+
+    Returns:
+        `numpy.ndarray`_: An array containing the 
+        object IDs of the objects linked together.
+        This array is aligned with sobjs
+    """
+    # Prep
+    norders = slit_left.shape[1]
+    slit_width = slit_righ - slit_left
+    nfound = len(sobjs)
+
+    #
+    FOF_frac = fof_link/(np.median(np.median(slit_width,axis=0)*plate_scale_ord))
+    # Run the FOF. We use fake coordinates
+    fracpos = sobjs.SPAT_FRACPOS
+    ra_fake = fracpos/1000.0  # Divide all angles by 1000 to make geometry euclidian
+    dec_fake = np.zeros_like(fracpos)
+    if nfound>1:
+        inobj_id, multobj_id, firstobj_id, nextobj_id \
+                = pydl.spheregroup(ra_fake, dec_fake, FOF_frac/1000.0)
+        # Modify to 1-based indexing
+        obj_id_init = inobj_id + 1
+    elif nfound==1:
+        obj_id_init = np.ones(1,dtype='int')
```

Comment:
Can `nfound` be 0?  If so, add an else here that throws an error?

Reply:
Good call